Core/Battler.gd

```diff
@@ -79,21 +79,35 @@ func _on_enemy_start_turn() -> void:
 	_handle_deaths()
 	
 	# generate list of enemy actions
-	var enemy_action_list: Array[EnemyAction] = []
-	
 	for enemy: Enemy in _enemy_list:
 		var enemy_attack: CardBase = enemy.get_behavior_component().attack
 		var enemy_action: EnemyAction = EnemyAction.new(enemy, enemy_attack, PlayerManager.player)
-		enemy_action_list.append(enemy_action)
-		
-	# execute enemy actions
-	for enemy_action: EnemyAction in enemy_action_list:
-		enemy_action.execute()
+		_enemy_action_list.append(enemy_action)
+	
+	# start the attack queue
+	assert(_enemy_action_list.size() >= 0, "Enemy tried to attack while there were attacks queued!")
+	_handle_enemy_attack_queue()
+
+
+func _handle_enemy_attack_queue() -> void:
+	var enemy_action: EnemyAction = _enemy_action_list[0]
+	_enemy_action_list.remove_at(0)
```

Comment:
is the `EnemyAction` passed as reference or as copy ?

Reply:
It should be a reference

Reply:
if it's a reference, what happens when you remove it from the list ? I think the list just drops the reference to the enemy action from its list, but it shouldn't set it to null I think ?

Reply:
Yeah it should just drop the reference from the list, shouldn't be set to null